application/ui/src/features/inspect/toolbar/sources/hooks/use-source-mutation.hook.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+import { $api } from '@geti-inspect/api';
+import { omit } from 'lodash-es';
+import { v4 as uuid } from 'uuid';
+
+import { SourceConfig } from '../util';
+
+export const useSourceMutation = (isNewSource: boolean) => {
+    const addSource = $api.useMutation('post', '/api/sources', {
+        meta: {
+            invalidates: [['get', '/api/sources']],
+        },
+    });
+    const updateSource = $api.useMutation('patch', '/api/sources/{source_id}', {
+        meta: {
+            invalidates: [['get', '/api/sources']],
+        },
+    });
+
+    return async (body: SourceConfig) => {
+        if (isNewSource) {
+            const sourcePayload = {
+                ...body,
+                id: uuid(),
```

Comment:
I would consider adding the `id` to this function's arguments. (which reading below should already be possible via `body.id`?)
That way you can do optimistic updates outside the scope of this function.

Alternatively assign it to a local variable `const id = uuid()` and then return it here instead of parse it out of the response (`String(response.id)`).